### PR TITLE
Use make_shared while creating shared_ptr

### PR DESCRIPTION
**Pull request by somnathr**
Because of less cpu cycles spent on creating shared_ptr , it is giving me ~5-8% performance boost on random write. Should be improving random read path as well , but not measured.

### CODE REVIEW

**Comment by yuyuyu101** [comment]
COOL

File: src/os/filestore/DBObjectMap.cc [hunk 1]
```diff
@@ -237,7 +237,7 @@ int DBObjectMap::DBObjectMapIteratorImpl::init()
       assert(0);
       return -EINVAL;
     }
-    parent_iter.reset(new DBObjectMapIteratorImpl(map, parent));
+    parent_iter = std::make_shared<DBObjectMapIteratorImpl>(map, parent);
```
Comment (tchaikov) [inline src/os/filestore/DBObjectMap.cc]: what's the advantage here?

File: src/osd/OSD.cc [hunk 2]
```diff
@@ -2725,8 +2725,8 @@ void OSD::recursive_remove_collection(ObjectStore *store, spg_t pgid, coll_t tmp
     coll_t(),
     make_snapmapper_oid());
 
-  ceph::shared_ptr<ObjectStore::Sequencer> osr(
-    new ObjectStore::Sequencer("rm"));
+  ceph::shared_ptr<ObjectStore::Sequencer> osr (std::make_shared<
```
Comment (tchaikov) [inline src/osd/OSD.cc]: might want to remove the space before `(`.

File: src/include/encoding.h [hunk 3]
```diff
@@ -595,7 +595,7 @@ inline void decode(std::vector<ceph::shared_ptr<T> >& v, bufferlist::iterator& p
   decode(n, p);
   v.resize(n);
   for (__u32 i=0; i<n; i++) {
-    v[i].reset(new T());
+    v[i] = std::make_shared<T>();
```
Comment (tchaikov) [inline src/include/encoding.h]: `reset(new T())` is better here, imho.

File: src/os/filestore/DBObjectMap.cc (hunk 1 above)
Comment (cbodley) [inline src/os/filestore/DBObjectMap.cc]: The advantage of make_shared() is that it combines the two allocations (the pointer itself, and the shared internal refcount/bookkeeping structure) into one. shared_ptr::reset() still needs to allocate this shared refcount structure, so the make_shared() optimization helps here too.

File: src/osd/OSD.cc (hunk 2 above)
Comment (tchaikov) [inline src/osd/OSD.cc]: nit, could replace the type declaration with an `auto`.

File: src/os/filestore/DBObjectMap.cc (hunk 1 above)
Comment (tchaikov) [inline src/os/filestore/DBObjectMap.cc]: @cbodley thanks for your explanation. i am reading EMC++ item21 now.

**Comment by somnathr** [comment]
@cbodley @tchaikov I would love to replace shared_ptr creation with make_shared for places like shared_cache, shared_registry and optracker but in those cases it is sending a deleter callback while creating shared_ptr and make_shared presently is not taking that parameter.

**Comment by athanatos** [comment]
Nice!

**Pull request merged**